airbyte-integrations/connectors/source-salesforce/source_salesforce/availability_strategy.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+import typing
+from typing import Optional, Tuple
+
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.availability_strategy import HttpAvailabilityStrategy
+from requests import HTTPError, codes
+
+if typing.TYPE_CHECKING:
+    from airbyte_cdk.sources import Source
+
+
+class SalesforceAvailabilityStrategy(HttpAvailabilityStrategy):
+    def handle_http_error(
+        self, stream: Stream, logger: logging.Logger, source: Optional["Source"], error: HTTPError
+    ) -> Tuple[bool, Optional[str]]:
+        """
+        There are several types of Salesforce sobjects that require additional processing:
+          1. Sobjects for which the user, after setting up the data using Airbyte, restricted access,
+             and we will receive 403 HTTP errors.
+          2. There are streams that do not allow you to make a sample using Salesforce `query` or `queryAll`.
+             And since we use a dynamic method of generating streams for Salesforce connector - at the stage of discover,
+             we cannot filter out these streams, so we catch them at the stage of reading data.
+        """
+        error_data = error.response.json()[0]
+        if error.response.status_code in [codes.FORBIDDEN, codes.BAD_REQUEST]:
+            error_code = error_data.get("errorCode", "")
+            if error_code != "REQUEST_LIMIT_EXCEEDED" or error_code == "INVALID_TYPE_FOR_OPERATION":
+                return False, f"Cannot receive data for stream '{stream.name}', error message: '{error_data.get('message')}'"
+        return True, None
```

Comment:
```suggestion
        raise e
```
If we hit an error we don't expect, we still want to raise it. That is how the [default availability strategy](https://github.com/airbytehq/airbyte/blob/1c482ee3393a5d6d4e6283095c605669015e5419/airbyte-cdk/python/airbyte_cdk/sources/streams/http/availability_strategy.py#L80-L81) handles it

Reply:
good catch, updated!